catenets/models/torch/snet.py

```diff
@@ -0,0 +1,342 @@
+import abc
+from typing import Any, Optional, Tuple
+
+import numpy as np
+import torch
+from torch import nn
+
+import catenets.logger as log
+from catenets.models.constants import (
+    DEFAULT_BATCH_SIZE,
+    DEFAULT_LAYERS_OUT,
+    DEFAULT_LAYERS_R,
+    DEFAULT_N_ITER,
+    DEFAULT_N_ITER_PRINT,
+    DEFAULT_NONLIN,
+    DEFAULT_PENALTY_L2,
+    DEFAULT_SEED,
+    DEFAULT_STEP_SIZE,
+    DEFAULT_UNITS_OUT,
+    DEFAULT_UNITS_R_BIG_S,
+    DEFAULT_VAL_SPLIT,
+)
+from catenets.models.torch.base import (
+    DEVICE,
+    BaseCATEEstimator,
+    BasicNet,
+    PropensityNet,
+    RepresentationNet,
+)
+from catenets.models.torch.utils.model_utils import make_val_split
+
+EPS = 1e-8
+
+
+class SNet(BaseCATEEstimator):
```

Comment:
Could we call this class something like 'BaseSNet' instead of 'SNet'? This is definitely my bad in terms of naming the estimators, but the most general SNet that I implemented (which bears the name SNet in my code) is different from this SNet.